### ISSUE TITLE
Code Cleanup Fixer -- Whitespace Normalization

### ISSUE THREAD

**Issue by kendrahavens** [dotnet/roslyn#37129]
[Original feedback on developer community](https://developercommunity.visualstudio.com/content/idea/545492/code-cleanup-fixer-whitespace-normalization.html)

The new Code Cleanup Fixers are extremely useful, and replace the need for most of the functionality in the popular extension Code Maid.  But one Fixer that is noticeably absent is whitespace normalization namely:

* Remove consecutive blank lines
* Normalize blank lines (e.g. every function has a single blank line after, braces don’t start or end in blank lines, etc)
* Trim spaces from the end of lines

And so on. You can already CTRL-K + D, but that doesn’t deal with a lot of whitespace problems.

**Comment by JadaVonRuth** [comment]
Absolutely amazing would be to create a way how to include other fixers from roslyn analyzers or stylecop etc. In order to be supported in code cleanup profiles. Also editorconfig could be useful for storing cleanup profiles for whole team so every dev will not have to construct same cleanup profile. Currently fixes which are not listed in cleanup list have to be used manually line by line, rule by rule which is time consuming process and could save a lot of time to every dev team.

**Comment by kendrahavens** [comment]
I agree, we should investigate automatically allowing any built-in or 3rd party fixer to be included in a code-cleanup profile. https://github.com/dotnet/roslyn/issues/40263

**Comment by killergege** [comment]
Agreed ! Automatically applying whitespace rules would be great.
Ideally, all formatting rules/category of rules from the editor config file or the VS configuration should have a fixer that would apply it automatically.

**Comment by CyrusNajmabadi** [comment]
This would be part of the formatting extensibility api i'm working on designing.
